### PR TITLE
Rename AToken/BToken to ALCA/ALCB

### PR DESCRIPTION
**Pull request by keirongulrajani**
## Scope
- Renamed AToken > ALCA
- Renamed BToken > ALCB
- Refactored interfaces for a/b tokens to staking/utility for improved clarity/maintenance

## Why?

[MP-728](https://madhive.atlassian.net/browse/MP-728)

## Todos

If any, what are the follow-up tasks required other than merging this PR? Have they been arranged?

- [ ] ???

### CODE REVIEW

File: bridge/contracts/ATokenBurner.sol [hunk 1]
```diff
@@ -1,15 +1,15 @@
 // SPDX-License-Identifier: MIT-open-group
 pragma solidity ^0.8.11;
 
-import "contracts/interfaces/IAToken.sol";
+import "contracts/interfaces/IStakingToken.sol";
 import "contracts/utils/ImmutableAuth.sol";
 
-/// @custom:salt ATokenBurner
+/// @custom:salt StakingTokenBurner
```
Comment (vtleonardo) [inline bridge/contracts/ATokenBurner.sol:7]: Let's not change the salt for now.

**Review by vtleonardo — CHANGES_REQUESTED** [review]

File: bridge/contracts/ATokenMinter.sol [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 pragma solidity ^0.8.11;
 
-import "contracts/interfaces/IAToken.sol";
+import "contracts/interfaces/IStakingToken.sol";
 import "contracts/utils/ImmutableAuth.sol";
 
 /// @custom:salt ATokenMinter
 /// @custom:deploy-type deployUpgradeable
-contract ATokenMinter is ImmutableAToken {
+contract ATokenMinter is ImmutableAToken, IStakingTokenMinter {
```
[the remark this answers is not in the document]
Comment (keirongulrajani) [reply bridge/contracts/ATokenMinter.sol:9]: Though technically it does not need to inherit the interface i believe its more for maintenance purposes as it has clear association with the `IStakingTokenMinter` interface which we refer to when calling the `mint` function elsewhere, so its more to ensure there is no question which contract is the true minter

File: bridge/contracts/interfaces/IUtilityToken.sol [hunk 3]
```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT-open-group
+pragma solidity ^0.8.0;
+
+interface IUtilityToken {
+    struct Deposit {
```
[the remark this answers is not in the document]
Comment (keirongulrajani) [reply bridge/contracts/interfaces/IUtilityToken.sol:5]: Good catch! I actually initially removed it from btoken but after reverting some changes to btoken i accidentally reverted this removal too!

File: bridge/contracts/ATokenMinter.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (vtleonardo) [reply bridge/contracts/ATokenMinter.sol:9]: It makes sense! Just make sure that ATokenBurner also inherits its interface for consistency's sake.

File: bridge/contracts/ATokenBurner.sol [hunk 4]
```diff
@@ -10,6 +10,6 @@ contract ATokenBurner is ImmutableAToken {
     constructor() ImmutableFactory(msg.sender) ImmutableAToken() {}
```
Comment (vtleonardo) [inline bridge/contracts/ATokenBurner.sol:10]: Please, add the inheritance of the IStakingTokenBurner interface here.

File: bridge/contracts/interfaces/IUtilityToken.sol (hunk 3 above)
Comment (vtleonardo) [inline bridge/contracts/interfaces/IUtilityToken.sol:5]: Can we move the struct to outside the interface?

File: bridge/contracts/ATokenMinter.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (keirongulrajani) [reply bridge/contracts/ATokenMinter.sol:9]: good catch! i thought i did put it there but i guess not, i'll address that now

**Review by nelsonhp — APPROVED** [review]

**Review by nelsonhp — APPROVED** [review]

**Review by vtleonardo — APPROVED** [review]

**Pull request merged**